script/plugin_webpack_directories.rb

```diff
@@ -27,12 +25,12 @@
 
   path = "#{dep.to_spec.full_gem_path}/webpack"
   entry = "#{path}/index.js"
-  # some plugins share the same base directory (tasks-core and tasks, REX etc)
   # skip the plugin if its path is already included
   next if config[:paths].include?(path)
-  if File.exist?(entry)
-    config[:entries][bundle_name dep.name] = entry
+  if File.exist?(entry) && dep.files.any? { |f| f == 'package.json' }
```

Comment:
Have you considered just checking if file exists same way we do it with index.js?

Reply:
If I understand it correctly this means it needs to be in the gemspec. I like that because otherwise we'll fail on it during packaging. Printing a warning if the file is present but not in the gemspec would be very developer friendly and prevent frustration so bonus points for that.

Also, wouldn't it be equal to `dep.files.include?('package.json')`?